bindings/rust/wasmedge-rs/src/config.rs

```diff
@@ -1,118 +1,17 @@
-use wasmedge_sys as ffi;
+use super::wasmedge;
 
+#[derive(Debug)]
 pub struct Config {
-    pub(crate) ctx: *mut ffi::WasmEdge_ConfigureContext,
-}
-
-impl Drop for Config {
-    fn drop(&mut self) {
-        unsafe { ffi::WasmEdge_ConfigureDelete(self.ctx) };
-    }
+    inner: wasmedge::Config,
 }
 
 impl Config {
-    pub fn new() -> Self {
-        Self::default()
-    }
-}
-
-impl Default for Config {
-    fn default() -> Self {
-        let ctx = unsafe { ffi::WasmEdge_ConfigureCreate() };
-        if ctx.is_null() {
-            panic!("failed to create WasmEdge configuration");
-        }
-        Self { ctx }
+    pub fn default() -> wasmedge::Config {
+        wasmedge::Config::default()
     }
-}
-
-macro_rules! impl_proposal_config {
-    ($( $proposal:ident ),+ $(,)+) => {
-        impl Config {
-            paste::paste! {
-                $(
-                    pub fn [<enable_ $proposal:lower:snake>](self, enable: bool) -> Self {
-                        let prop = ffi::[<WasmEdge_Proposal_ $proposal>];
-                        if enable {
-                            unsafe { ffi::WasmEdge_ConfigureAddProposal(self.ctx, prop) };
-                        } else {
-                            unsafe { ffi::WasmEdge_ConfigureRemoveProposal(self.ctx, prop) };
-                        }
-                        self
-                    }
 
-                    pub fn [<has_ $proposal:lower:snake>](self) -> bool {
-                        let prop = ffi::[<WasmEdge_Proposal_ $proposal>];
-                        unsafe { ffi::WasmEdge_ConfigureHasProposal(self.ctx, prop) }
-                    }
-                )+
-            }
-        }
+    pub fn with_wasi() -> wasmedge::Config {
+        let config = Self::default();
+        config.reg_wasi()
```

Comment:
This method is defined by you. I just followed the method you defined. I guess it means register wasi, because the corresponding c-api is `WasmEdge_HostRegistration_Wasi`.

Reply:
`enable_wasi ` is good.